src/main/java/org/opensearch/knn/index/util/IndexHyperParametersUtil.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.index.util;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import lombok.extern.log4j.Log4j2;
+import org.opensearch.Version;
+import org.opensearch.knn.index.KNNSettings;
+
+/**
+ * This class acts as an abstraction to get the default hyperparameter values for different parameters used in the
+ * Nearest Neighbor Algorithm across different version of Index.
+ */
+@Log4j2
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class IndexHyperParametersUtil {
+
+    private static final int INDEX_KNN_DEFAULT_ALGO_PARAM_EF_CONSTRUCTION_OLD_VALUE = 512;
+    private static final int INDEX_KNN_DEFAULT_ALGO_PARAM_EF_SEARCH_OLD_VALUE = 512;
+
+    /**
+     * Returns the default value of EF Construction that should be used for the input index version. After version 2.12.0
+     * of Opensearch we are have reduced the value of ef_construction in favor of better build times.
+     *
+     * @param indexVersion {@code Version} of the index with which it was created.
+     * @return default value of EF Construction that should be used for the input index version.
+     */
+    public static int getHNSWEFConstructionValue(@NonNull final Version indexVersion) {
```

Comment:
Do we need one method per parameter, or could we build a map and then require the user to pass in the algo and parameter names and we do a lookup?

Reply:
Having a map is more error prone, as construction of map can lead to issues and more branches. Right now this is a simple way. Going forward we can evolve this logic and functions to be more generic if required. As it is not required right now we should not add.